### PR TITLE
Phase2-hgx101 Try to introduce the new geometry of HGCal - only done up to EE so far

### PR DESCRIPTION
**Pull request by bsunanda**
Introduce the creation of wafers with the new layout and the layers for the EE part

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by davidlange6** [comment]
please test

File: Geometry/HGCalCommonData/plugins/DDHGCalCell.cc [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+#include "DetectorDescription/Core/interface/DDCurrentNamespace.h"
+#include "DetectorDescription/Core/interface/DDLogicalPart.h"
+#include "DetectorDescription/Core/interface/DDMaterial.h"
+#include "DetectorDescription/Core/interface/DDSolid.h"
+#include "DetectorDescription/Core/interface/DDSplit.h"
+#include "DetectorDescription/Core/interface/DDutils.h"
+#include "Geometry/HGCalCommonData/plugins/DDHGCalCell.h"
+#include "CLHEP/Units/GlobalSystemOfUnits.h"
+
+#define EDM_ML_DEBUG
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/plugins/DDHGCalCell.cc]: disable debug flag

File: Geometry/HGCalCommonData/plugins/DDHGCalCell.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#ifndef HGCalCommonData_DDHGCalCell_h
+#define HGCalCommonData_DDHGCalCell_h
+
+#include <map>
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/plugins/DDHGCalCell.h]: remove unneeded include

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by davidlange6** [comment]
please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna @kpedro88 Could you approve this PR?

**Comment by kpedro88** [comment]
+1

**Comment by bsunanda** [comment]
@ianna @civanch Please approve this. It is the first step for new geometry of HGCal. I have to do this in steps because of complexity of the new design

**Comment by ianna** [comment]
+1

**Comment by fabiocos** [comment]
@bsunanda @kpedro88 is this code practically run in one of the tested upgrade scenarios, or are we just testing that it doesn't break DDHGcalEEAlgo (as I guess, since the v9 version is not present up to now, and I do not see associated changes in the test configurations)? As a starting point development for Phase2 it could be ok, but for my understanding I would like a confirmation

**Comment by kpedro88** [comment]
@fabiocos the latter - just an intermediate PR to build up for further development

**Comment by bsunanda** [comment]
@fabiocos - we cannot have a full upgrade scenario with this for a while - first to complete the G4 geometry part, then redefine DetID's, then interface with SIM step, then the reconstruction geometry and topology and finally tuning the digitization, local reco, particle flow code + validation & DQM. So it is a long path

**Comment by bsunanda** [comment]
@fabiocos @kpedro88 This will not break any workflow currency available

**Comment by fabiocos** [comment]
+1

**Pull request merged**